odl/trafos/wavelet.py

```diff
@@ -652,21 +982,28 @@ def _call(self, x):
             The length of the array depends on the size of input image to
             be transformed and on the chosen wavelet basis.
         """
-        if x.space.ndim == 1:
-            coeff_list = pywt.wavedec(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 2:
-            coeff_list = pywt.wavedec2(x, self.wbasis, self.mode, self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_list, self.size_list)
-            return self.range.element(coeff_arr)
-
-        if x.space.ndim == 3:
-            coeff_dict = wavelet_decomposition3d(x, self.wbasis, self.mode,
-                                                 self.nscales)
-            coeff_arr = pywt_coeff_to_array(coeff_dict, self.size_list)
-
+        if self._axes is None:
+            if x.space.ndim == 1:
+                coeff_list = pywt.wavedec(x, self.wbasis, self.mode,
+                                          self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 2:
+                coeff_list = pywt.wavedec2(x, self.wbasis, self.mode,
+                                           self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+                return self.range.element(coeff_arr)
+
+            if x.space.ndim == 3:
+                coeff_list = wavelet_decomposition3d(x, self.wbasis, self.mode,
+                                                     self.nscales)
+                coeff_arr = pywt_list_to_array(coeff_list, self.size_list)
+
+                return self.range.element(coeff_arr)
+        else:
+            coeff_dict = pywt.dwtn(x, self.wbasis, self.mode, self._axes)
```

Comment:
Of course,  the sizes are dependent on the chosen boundary behaviour and the chosen wavelet basis...

> Do the PyWavelets guys have a document lying around that explains what exactly is computed? Something like this?

Can't say off-hand, but I'll try to look for a one.